lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8084,6 +8086,25 @@ bool swift::importer::isMutabilityAttr(const clang::SwiftAttrAttr *swiftAttr) {
          swiftAttr->getAttribute() == "nonmutating";
 }
 
+static bool importAsUnsafe(const ASTContext &context,
+                           const clang::RecordDecl *decl,
+                           const Decl *MappedDecl) {
+  if (!context.LangOpts.hasFeature(Feature::SafeInterop) ||
+      !context.LangOpts.hasFeature(Feature::AllowUnsafeAttribute) || !decl)
+    return false;
+
+  if (const auto *classDecl = dyn_cast<ClassDecl>(MappedDecl)) {
+    if (classDecl->getForeignClassKind() == ClassDecl::ForeignKind::CFType)
+      return false;
+  }
```

Comment:
Everything that is imported as a Swift class is safe, so I think the entire condition here is `isa<ClassDecl>(MappedDecl)`.

Reply:
Other examples of things that are safe, e.g., https://github.com/swiftlang/swift/pull/76408

Reply:
> Everything that is imported as a Swift class is safe
I think this might depend on the definition of safety. In the meantime, I realized, this might not be true for CF types either.

For example, `CFArray` itself is safe, because it is properly refcounted, but it might have non-escapable elements. So a function returning a `CFArray` is not safe without explicit annotations attesting that the returned value is self contained. What do you think?

Reply:
We talked about this offline. Probably the good direction here is expect the users to explicitly annotate types that are reference counted and not escapable. Reference counted types most of the times should only rely on reference counting to make sure the lifetimes are right, so it is a strong signal that the type is safe to escape.